packages/payment-backend/src/types.ts

```diff
@@ -73,3 +73,13 @@ export interface Payment {
   token: string
   timestamp: string
 }
+
+export interface PaymentFilter {
+  filterId?: string
+  createdAt?: string
+  namespace: string
+  referenceId: string
+  referenceType: string
```

Comment:
Can this be enum like variable?

`referenceType: 'order' | 'merchant'`

Reply:
I believe so, I'll add the changes.

Reply:
I leveraged enums (ReferenceType and FilterType) to ensure the specific values.